src/middlewared/middlewared/plugins/cluster_linux/ctdb_private.py

```diff
@@ -0,0 +1,138 @@
+from middlewared.schema import List, Dict, IPAddr, Int
+from middlewared.service import accepts, private, job, CRUDService, ValidationErrors
+import middlewared.sqlalchemy as sa
+from .utils import CTDBConfig, JOB_LOCK
+
+
+PRIVATE_IP_FILE = CTDBConfig.PRIVATE_IP_FILE.value
+
+
+class CtdbPrivateIpModel(sa.Model):
+    __tablename__ = 'ctdb_private_ips'
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+    ip = sa.Column(sa.String(45), nullable=True)
+
+
+class CtdbPrivateIpService(CRUDService):
+
+    class Config:
+        namespace = datastore = 'ctdb.private.ips'
+
+    @private
+    async def common_validation(self, data, schema_name, verrors):
+
+        # make sure something was given to us
+        if not data['private_ips']:
+            verrors.add(
+                f'{schema_name}.no_ips',
+                'Private IP(s) must be specified.',
+            )
+
+        verrors.check()
+
+        # normalize the data and remove netmask info (if its given to us)
+        # from `ip` keys since it's not used
+        for i in data['private_ips']:
+            i['ip'] = i['ip'].split('/')[0]
+
+        if schema_name == 'node_create':
```

Comment:
If we have specific create validation, seems it should not be part of the common validation method?